test/e2e/csv_e2e_test.go

```diff
@@ -988,61 +967,61 @@ var _ = Describe("CSV", func() {
 		}()
 
 		fetchedCSV, err := fetchCSV(crc, csv.Name, testNamespace, csvSucceededChecker)
-		require.NoError(GinkgoT(), err)
+		Expect(err).ShouldNot(HaveOccurred())
 
 		// Should create Deployment
 		dep, err := c.GetDeployment(testNamespace, depName)
-		require.NoError(GinkgoT(), err, "error getting expected Deployment")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected Deployment")
 
 		// Should create APIService
 		apiService, err := c.GetAPIService(apiServiceName)
-		require.NoError(GinkgoT(), err, "error getting expected APIService")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected APIService")
 
 		// Should create Service
 		serviceName := fmt.Sprintf("%s-service", depName)
 		_, err = c.GetService(testNamespace, serviceName)
-		require.NoError(GinkgoT(), err, "error getting expected Service")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected Service")
 
 		// Should create certificate Secret
 		secretName := fmt.Sprintf("%s-cert", serviceName)
 		_, err = c.GetSecret(testNamespace, secretName)
-		require.NoError(GinkgoT(), err, "error getting expected Secret")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected Secret")
 
 		// Should create a Role for the Secret
 		_, err = c.GetRole(testNamespace, secretName)
-		require.NoError(GinkgoT(), err, "error getting expected Secret Role")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected Secret Role")
 
 		// Should create a RoleBinding for the Secret
 		_, err = c.GetRoleBinding(testNamespace, secretName)
-		require.NoError(GinkgoT(), err, "error getting exptected Secret RoleBinding")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting exptected Secret RoleBinding")
 
 		// Should create a system:auth-delegator Cluster RoleBinding
 		_, err = c.GetClusterRoleBinding(fmt.Sprintf("%s-system:auth-delegator", serviceName))
-		require.NoError(GinkgoT(), err, "error getting expected system:auth-delegator ClusterRoleBinding")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected system:auth-delegator ClusterRoleBinding")
 
 		// Should create an extension-apiserver-authentication-reader RoleBinding in kube-system
 		_, err = c.GetRoleBinding("kube-system", fmt.Sprintf("%s-auth-reader", serviceName))
-		require.NoError(GinkgoT(), err, "error getting expected extension-apiserver-authentication-reader RoleBinding")
+		Expect(err).ShouldNot(HaveOccurred(), "error getting expected extension-apiserver-authentication-reader RoleBinding")
 
 		// Store the ca sha annotation
 		oldCAAnnotation, ok := dep.Spec.Template.GetAnnotations()[install.OLMCAHashAnnotationKey]
-		require.True(GinkgoT(), ok, "expected olm sha annotation not present on existing pod template")
+		Expect(ok).Should(BeTrue(), "expected olm sha annotation not present on existing pod template")
 
 		// Induce a cert rotation
 		now := metav1.Now()
 		fetchedCSV.Status.CertsLastUpdated = &now
```

Comment:
a rather frequent flake happens here:    	
```
Error Trace:	csv_e2e_test.go:1036
  	            				runner.go:113
  	            				runner.go:64
  	            				it_node.go:26
  	            				spec.go:215
  	            				spec.go:138
  	            				spec_runner.go:200
  	            				spec_runner.go:170
  	            				spec_runner.go:66
  	            				suite.go:62
  	            				ginkgo_dsl.go:226
  	            				ginkgo_dsl.go:214
  	            				e2e_test.go:54
  	Error:      	Received unexpected error:
  	            	Operation cannot be fulfilled on clusterserviceversions.operators.coreos.com "hat-server45hc6": the object has been modified; please apply your changes to the latest version and try again 
```
Maybe fetch CSV right before updating it.

https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/origin-ci-test/pr-logs/pull/operator-framework_operator-lifecycle-manager/1592/pull-ci-operator-framework-operator-lifecycle-manager-master-e2e-aws-olm/1275777878046281728

Reply:
https://github.com/operator-framework/operator-lifecycle-manager/pull/1604

Reply:
this is addressed in #1604